pkg/storagev2/migration/migration.go

```diff
@@ -0,0 +1,68 @@
+package migration
+
+import (
+	"encoding/binary"
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"unsafe"
+)
+
+type Version = uint64
+type StepFn func(storage.Store) error
+type StepsMap = map[Version]StepFn
+
+func Migrate(s storage.Store, sm StepsMap) error {
+	version, err := GetVersion(s)
+	if err != nil {
+		return err
+	}
+	for key, step := range sm {
+		if key == version+1 {
+			err := step(s)
+			if err != nil {
+				return err
+			}
+			err = s.Put(&storageVersionItem{Version: key})
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
this for cycle will execute only once, but it needs to execute as many as there are newer versions